yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClusterSchedulerBackend.scala

```diff
@@ -37,9 +38,12 @@ private[spark] class YarnClusterSchedulerBackend(
     scheduler: TaskSchedulerImpl,
     sc: SparkContext)
   extends YarnSchedulerBackend(scheduler, sc) {
+  private val services: YarnExtensionServices = new YarnExtensionServices()
```

Comment:
It's in the core module, which doesn't require hadoop 2.x; the patch as it is today keeps everything self-contained into the spark-yarn package.